app/src/routes/register/register-form.vue

```diff
@@ -0,0 +1,126 @@
+<script setup lang="ts">
+import api, { RequestError } from '@/api';
+import { translateAPIError } from '@/lang';
+import { ErrorCode } from '@directus/errors';
+import { computed, ref } from 'vue';
+import { useI18n } from 'vue-i18n';
+
+type Credentials = {
+	email: string;
+	password: string;
+};
+
+const { t } = useI18n();
+const isLoading = ref(false);
+const email = ref<string | null>(null);
+const password = ref<string | null>(null);
+const passwordVerification = ref<string | null>(null);
+const error = ref<RequestError | string | null>(null);
+const passwordsMatch = computed(() => password.value === passwordVerification.value);
+
+const emit = defineEmits<{
+	wasSuccessful: [boolean];
+}>();
+
+const errorFormatted = computed(() => {
+	// Show "Wrong username or password" for wrongly formatted emails as well
+	if (error.value === ErrorCode.InvalidPayload) {
+		return translateAPIError(ErrorCode.InvalidCredentials);
+	}
```

Comment:
This message makes sense for login, but I think for registering it should be: Invalid username or password.